rectools/visuals/metrics_app.py

```diff
@@ -0,0 +1,158 @@
+import typing as tp
+
+import ipywidgets as widgets
+import pandas as pd
+import plotly.express as px
+import plotly.graph_objs as go
+from IPython.display import display
+
+from rectools import Columns
+
+DEFAULT_WIDTH = 800
+DEFAULT_HEIGHT = 600
+
+
+class MetricsApp:
+    """
+    Jupyter widgets app for metric visualization and comparison.
+    Do not create instances of this class directly. Use `MetricsApp.construct` method instead.
+    """
+
+    def __init__(
+        self,
+        models_metrics: pd.DataFrame,
+        show_legend: bool = True,
+        auto_display: bool = True,
+        plotly_kwargs: tp.Optional[tp.Dict[str, tp.Any]] = None,
+    ):
+        self.models_metrics = models_metrics
+        self.show_legend = show_legend
+        self.auto_display = auto_display
+        self.plotly_kwargs = plotly_kwargs if plotly_kwargs is not None else {}
+
+        self.fig = go.Figure()
+
+        self._validate_input_data()
+        if self.auto_display:
+            self.display()
+
+    @classmethod
+    def construct(
+        cls,
+        models_metrics: pd.DataFrame,
+        show_legend: bool = True,
+        auto_display: bool = True,
+        plotly_kwargs: tp.Optional[tp.Dict[str, tp.Any]] = None,
+    ) -> "MetricsApp":
+        """Construct a widget-based application for visualizing metrics from the provided dataframe.
+
+        Parameters
+        ----------
+        models_metrics : pd.DataFrame
+            A pandas DataFrame containing metrics for visualization. Required columns:
+                - `Columns.Models` - model names
+                - `Columns.Split` - fold number
+                - Any other numeric columns which represent metric values
+        show_legend : bool, default True
+            Specifies whether to display the chart legend.
+        auto_display : bool, default True
+            Automatically displays the widgets immediately after initialization.
+        plotly_kwargs : tp.Optional[tp.Dict[str, tp.Any]], optional, default None
+            Additional arguments for customizing the `px.scatter` plot.
+
+        Returns
```

Comment:
It also would be great to add `Examples` here with one simple example. This will be rendered in documentation just like here: https://rectools.readthedocs.io/en/stable/api/rectools.visuals.visual_app.VisualApp.html